src/propslib/DelegationConfig.cpp

```diff
@@ -307,23 +307,64 @@ CATCH_RETURN()
     return S_OK;
 }
 
-[[nodiscard]] HRESULT DelegationConfig::s_Set(PCWSTR value, const CLSID clsid) noexcept
+[[nodiscard]] HRESULT DelegationConfig::s_Set(PCWSTR value, const CLSID clsid, const bool useRegExe) noexcept
 try
 {
-    wil::unique_hkey currentUserKey;
-    wil::unique_hkey consoleKey;
+    // BODGY
+    // A Centennial application is not allowed to write the system registry and is redirected
+    // to a per-package copy-on-write hive.
+    // The restricted capability "unvirtualizedResources" can be combined with
+    // desktop6:RegistryWriteVirtualization to opt-out... but...
+    // - It will no longer be double-click installable through the App Installer
+    // - It requires a special exception to submit to the store
+    // - There MAY be some cleanup logic where the app catalog may try to undo
+    //   whatever the package did.
+    // This works around it by shelling out to reg.exe because somehow that's just peachy.
+    if (useRegExe)
+    {
+        wil::unique_cotaskmem_string str;
+        RETURN_IF_FAILED(StringFromCLSID(clsid, &str));
+
+        auto regExePath = wil::ExpandEnvironmentStringsW<std::wstring>(L"%WINDIR%\\System32\\reg.exe");

```

Comment:
Technically we can `GetSystemDirectory` (for which wil has a wrapper) -- might be (delta) "easier"

Reply:
That sounds like making one more call and concatenate.... I think I'll just leave it this way especially since I stole this from another part of our codebase already (and subbed the exe name)